### ISSUE TITLE
wrong code for uncertain function parameters

### ISSUE THREAD

**Issue by GoogleCodeExporter** [google-code-export/asdec#6]
```
as source code fragment :

public function test( ... rest ) : void 
{
}

compile to swf then decompile to as source 

public function test(, ... rest) : void
{
         return;
}

something is wrong in methodinfo




```

Original issue reported on code.google.com by `Wo.Bian....@gmail.com` on 19 Dec 2011 at 11:04

**Comment by GoogleCodeExporter** [comment]
```
i have found the bug exist in the following source code fragment in file 
.../asdec/abc/types/MethodInfo.java

if(flagNeed_rest())
{
      // if paramStr == "" 
      // there shoud not have a ',' 
      paramStr+=", ... ";
      if(!localRegNames.isEmpty())
      {
           paramStr+=localRegNames.get(param_types.length+1);
      }else{
          paramStr+="rest";
      }

}

```

Original comment by `Wo.Bian....@gmail.com` on 19 Dec 2011 at 11:36

**Comment by GoogleCodeExporter** [comment]
```
   if(params.length == 0 )
       paramStr += "..." ;
   else
       paramStr += ", ...";

```

Original comment by `Wo.Bian....@gmail.com` on 19 Dec 2011 at 11:46

**Comment by GoogleCodeExporter** [comment]
Original comment by `jindra.p...@gmail.com` on 27 Dec 2011 at 7:44
- Changed state: **Started**
- Added labels: **Component-Logic**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision c764b1e4f9f5.
```

Original comment by `jindra.p...@gmail.com` on 27 Dec 2011 at 2:14
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Thank you very much for reporting.
Problem fix is in the repository.
You can try it yourself if you compile it.
```

Original comment by `jindra.p...@gmail.com` on 27 Dec 2011 at 2:20
